### DISCUSSION TITLE
The overall shift in energy

### DISCUSSION THREAD

**Discussion by liubaoshuai1402** [ACEsuit/mace#762] [Q&A]
Dear developers,

I counter some problems and am confused. The following picture is helpful for you to know what the problem is.
![Figure_1](https://github.com/user-attachments/assets/9eee6a1a-9270-41a0-9001-a7a281dcb56f)
**The total energy per atom calculates by mace model is always slightly higher than vasp by a Fixed value ！**
Here are my MACE0.39 parameters:

tmux new -s my_session -d "mace_run_train \
    --name=MACE_model \
    --seed=123 \
    --log_dir=MACE_model \
    --model_dir=MACE_model \
    --checkpoints_dir=MACE_model \
    --results_dir=MACE_model \
    --device=cuda \
    --train_file=train.xyz \
    --valid_fraction=0.1 \
    --test_file=test.xyz \
    --compute_forces=True \
    --compute_stress=True \
    --energy_key=energy_vasp \
    --forces_key=forces_vasp \
    --stress_key=stress_vasp \
    --E0s='isolated' \
    --hidden_irreps='128x0e + 128x1o' \
    --r_max=5.0 \
    --batch_size=4 \
    --max_num_epochs=600 \
    --swa \
    --start_swa=420 \
    --ema \
    --ema_decay=0.99 \
    --default_dtype=float64 \
    --save_cpu \
    --lr=0.01 \
    --scaling=rms_forces_scaling \
    --multiheads_finetuning=False"

I guess this is caused by my training set which contains similar configurations but different charge.
(That is to say, for example, hydrogen enter SiO2 by the format of proton.
A perfect cell contains 100 atoms of Si and O. I add one H then delete one electron to study the behavior of proton with Si and O. Namely,101 atoms and 1 charge. I add two H then delete two electrons to study the behavior of proton and proton. Namely, 102 atoms and 2 charges.)

I also notice a similar discussion, https://github.com/ACEsuit/mace/discussions/579, it seems that the overall shift of energy can be caused by the poor ability of ML potential at extrapolating in density, **I can not understand 'density' well**.
Is it meant that I should use different-size supercell (2 x 2 x 2 and 3 x 3 x 3 ?) or give some stress to the cell (namely Artificially stretching or compressing a lattice vector by 3% 5% and so on), I have added the data for the latter method to the training set, in fact, it is necessary when it comes to study the mechanical properties of bulk phase.

Here is error table:

2024-12-24 08:03:39.637 INFO: Error-table on TRAIN and VALID:
+---------------+---------------------+------------------+-------------------+
|  config_type  | RMSE E / meV / atom | RMSE F / meV / A | relative F RMSE % |
+---------------+---------------------+------------------+-------------------+
| train_default |            0.9                  |         13.2            |          0.96             |
| valid_default |            1.0                  |         39.1            |          2.75             |
+---------------+---------------------+------------------+-------------------+
2024-12-24 08:03:39.638 INFO: Evaluating Default_default ...
2024-12-24 08:03:40.435 INFO: Error-table on TEST:
+-----------------+---------------------+------------------+-------------------+
|   config_type   | RMSE E / meV / atom | RMSE F / meV / A | relative F RMSE % |
+-----------------+---------------------+------------------+-------------------+
| Default_default |            7.5                 |         39.8            |        202.51          |
+-----------------+---------------------+------------------+-------------------+

Since the forces prediction seems look OK, should I neglect this problem? 
Any advice would be appreciated.
Thanks!

**Comment by ilyes319** [comment]
Hello @liubaoshuai1402, could you please share you training log files here so I can check different things?
Note that we have a second training stage that should help with that, just add `--stage_two` to your input.
